text/0000-integer_atomics.md

```diff
@@ -0,0 +1,105 @@
+- Feature Name: integer_atomics
+- Start Date: 2016-03-14
+- RFC PR: (leave this empty)
+- Rust Issue: (leave this empty)
+
+# Summary
+[summary]: #summary
+
+This RFC basically changes `core::sync::atomic` to look like this:
+
+```rust
+#[cfg(target_has_atomic = "8")]
+struct AtomicBool {}
+#[cfg(target_has_atomic = "8")]
+struct AtomicI8 {}
+#[cfg(target_has_atomic = "8")]
+struct AtomicU8 {}
+#[cfg(target_has_atomic = "16")]
+struct AtomicI16 {}
+#[cfg(target_has_atomic = "16")]
+struct AtomicU16 {}
+#[cfg(target_has_atomic = "32")]
+struct AtomicI32 {}
+#[cfg(target_has_atomic = "32")]
+struct AtomicU32 {}
+#[cfg(target_has_atomic = "64")]
+struct AtomicI64 {}
+#[cfg(target_has_atomic = "64")]
+struct AtomicU64 {}
+#[cfg(target_has_atomic = "128")]
+struct AtomicI128 {}
+#[cfg(target_has_atomic = "128")]
+struct AtomicU128 {}
+#[cfg(target_has_atomic = "ptr")]
+struct AtomicIsize {}
+#[cfg(target_has_atomic = "ptr")]
+struct AtomicUsize {}
+#[cfg(target_has_atomic = "ptr")]
+struct AtomicPtr<T> {}
```

Comment:
@jmesmon I think the ship has sailed on supporting ptr-sized atomics conditionally. We already guarantee support on platforms that don't have atomics.

This is a pretty bad situation though because our atomics are not guaranteed [to be lock free](https://github.com/rust-lang/rust/issues/32967).

Saving grace may be that no architectures without atomics are tier-1 so *maybe* we can roll back these prior decisions.